in_toto/verifylib.py

```
@@ -239,16 +239,17 @@ def verify_all_steps_signatures(layout, chain_link_dict):
 
   """
   for step in layout.steps:
-    keys_dict = {}
-
     # Find the according link for this step
     key_link_dict = chain_link_dict[step.name]
 
-    # Create the dictionary of keys for this step
-    for pubkeyid in step.pubkeys:
-      keys_dict[pubkeyid] = layout.keys[pubkeyid]
-
     for keyid, link in six.iteritems(key_link_dict):
+      keys_dict = {}
+
+      # Create the dictionary of keys for this step
+      for pubkeyid in step.pubkeys:
+          if pubkeyid == keyid:
+            keys_dict[pubkeyid] = layout.keys[pubkeyid]
```

Here are some notes,
- make sure to always have consistent indentation (2 spaces per level, 4 for line continuation),
- use `if pubkeyid in step.pubkeys` instead of the loop,
- I think it makes sense to fail early (plus give user feedback) , e.g. something like:
```python
if pubkeyid in step.pubkeys:
  keys_dict[pubkeyid] = layout.keys[pubkeyid]
else:
  raise SignatureVerificationError("<some descriptive message here>")
```
- also remove the signature verification part from the [threshold verification function]( https://github.com/in-toto/in-toto/blob/develop/in_toto/verifylib.py#L986-L993) as mentioned in the related [issue](https://github.com/in-toto/in-toto/issues/82).
- try to add some comments that describe what your changes are good for,
- if you fix an issue please mention the issue in the PR.

---

I'm adding to this, where does the keyid prefix of the link filename check actually happen?  I'm finding it hard to parse.

---

The variable value `keyid` holds the (whole) keyid, determined by the filename infix. So instead of taking all keys authorized for a certain step (`step.pubkeys`) and verifying that the link file is signed by one ore more of those keys, we take the key, identified by `keyid`, check if it is authorized, and then verify if the link was signed with that key.

---

> if pubkeyid in step.pubkeys:
  keys_dict[pubkeyid] = layout.keys[pubkeyid]
else:
  raise SignatureVerificationError("<some descriptive message here>")


@lukpueh  Do we need this? This will raise an error if the keyid (as in the filename) is not present in the layout. But if it isn't present in the layout, our initial load_metadata will fail.
Or maybe I'm missing something.

---

Isn't this check something you are [proposing in your PR](https://github.com/RohanBhirangi/in-toto/blob/225c0169f672fa16972ba893c67a39b149313343/in_toto/verifylib.py#L249-L250)? 

The only difference is that you would let a non-authorized key fail implicitly in the `Signable`'s `verify_signature` function due to an [empty signatures field](https://github.com/in-toto/in-toto/blob/develop/in_toto/models/common.py#L102) or an [empty key dictionary](https://github.com/in-toto/in-toto/blob/develop/in_toto/models/common.py#L108), and I suggest to let it fail explicitly. 

Maybe `SignatureVerificationError` is not the best exception type, maybe it should be something like `AuthorizationError`. 

In any case I'd say it does not hurt to re-check the authorization here, although it's already done in the `import_step_metadata_from_files_as_dict` function.